backend/telemetry_core/src/feed_message.rs

```diff
@@ -183,20 +186,21 @@ impl FeedMessageWrite for AddedNode<'_> {
         let AddedNode(nid, node, expose_node_details) = self;
 
         let details = node.details();
-        // Hide the ip, sysinfo and hwbench if the `expose_node_details` flag was not specified.
+        // Always include sysinfo, conditionally include ip and hwbench based on expose_node_details.
         let node_hwbench = node.hwbench();
-        let (ip, sys_info, hwbench) = if *expose_node_details {
-            (&details.ip, &details.sysinfo, &node_hwbench)
-        } else {
-            (&None, &None, &None)
-        };
+        let ip = if *expose_node_details { &details.ip } else { &None };
+        let sys_info = &details.sysinfo;
```

Comment:
Is there a reason why this should no longer respect the `--expose-node-details` flag? I think the point of this flag is to provide some privacy for nodes on the default telemetry instance, since they will connect to it by default. Then, on eg the 1kv telemetry instance we allow node details to be exposed since nodes must opt in to sharing telemetry there anyway.

It's debatable whether this privacy is required or not for the system info though. Perhaps there is a case for hiding IPs but exposing the hardware details, but I'd want to ask around a bit on this and gather some thoughts.